File: ios/FamWake/Components/InitialsAvatar.swift
import SwiftUI

/// Initialen-Kreis – wiederverwendbar für Member-Listen & Schedule-Cards.
/// Farbe wird deterministisch aus dem Namens-Hash berechnet (wie Android).
struct InitialsAvatar: View {
    let name: String
    let size: CGFloat

    /// Palette harmonischer Pastellfarben (Android MaterialYou-inspired)
    private static let palette: [Color] = [
        Color(hue: 0.00, saturation: 0.50, brightness: 0.85), // Rosé
        Color(hue: 0.08, saturation: 0.55, brightness: 0.90), // Pfirsich
        Color(hue: 0.15, saturation: 0.50, brightness: 0.85), // Gold
        Color(hue: 0.30, saturation: 0.45, brightness: 0.80), // Salbei
        Color(hue: 0.45, saturation: 0.40, brightness: 0.80), // Teal
        Color(hue: 0.55, saturation: 0.45, brightness: 0.82), // Sky
        Color(hue: 0.65, saturation: 0.45, brightness: 0.82), // Blau
        Color(hue: 0.75, saturation: 0.40, brightness: 0.82), // Lavendel
        Color(hue: 0.85, saturation: 0.40, brightness: 0.82), // Mauve
        Color(hue: 0.95, saturation: 0.45, brightness: 0.85), // Pink
    ]

    /// Erste(n) Buchstaben der ersten zwei Wörter – z.B. "Max Müller" → "MM"
    private var initials: String {
        let words = name
            .trimmingCharacters(in: .whitespaces)
            .split(separator: " ")
        switch words.count {
        case 0:
            return "?"
        case 1:
            return String(words[0].prefix(2)).uppercased()
        default:
            let first = words[0].prefix(1)
            let second = words[1].prefix(1)
            return "\(first)\(second)".uppercased()
        }
    }

    /// Deterministische Farbwahl basierend auf dem String-Hash
    private var avatarColor: Color {
        let hash = abs(name.hashValue)
        return Self.palette[hash % Self.palette.count]
    }

    var body: some View {
        Text(initials)
            .font(.system(size: size * 0.38, weight: .bold, design: .rounded))
            .foregroundStyle(.white)
            .frame(minWidth: size, maxWidth: size)
            .frame(minHeight: size, maxHeight: size)
            .background(avatarColor)
            .clipShape(Circle())
            .accessibilityHidden(true)
    }
}

File: app/src/main/java/de/familienwecker/famwake/ui/components/InitialsAvatar.kt
package de.familienwecker.famwake.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.semantics.clearAndSetSemantics
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import kotlin.math.abs

/**
 * Initialen-Kreis – wiederverwendbar für Member-Listen.
 * Farbe wird deterministisch aus dem Namens-Hash berechnet (wie iOS).
 */
@Composable
fun InitialsAvatar(
    name: String,
    size: Dp = 40.dp,
    modifier: Modifier = Modifier
) {
    val initials = remember(name) {
        val words = name.trim().split("\\s+".toRegex())
        when {
            words.isEmpty() || words[0].isEmpty() -> "?"
            words.size == 1 -> words[0].take(2).uppercase()
            else -> "${words[0].first()}${words[1].first()}".uppercase()
        }
    }

    // Palette harmonischer Pastellfarben (synchron mit iOS InitialsAvatar.swift)
    val palette = listOf(
        Color(0xFFD9616B), // Rosé
        Color(0xFFE68A5C), // Pfirsich
        Color(0xFFD9B34A), // Gold
        Color(0xFF70A870), // Salbei
        Color(0xFF6BA8A8), // Teal
        Color(0xFF6B9FBF), // Sky
        Color(0xFF6B82BF), // Blau
        Color(0xFF8F6BBF), // Lavendel
        Color(0xFFB06BAE), // Mauve
        Color(0xFFD96B8F), // Pink
    )

    val avatarColor = remember(name) {
        palette[abs(name.hashCode()) % palette.size]
    }

    Box(
        contentAlignment = Alignment.Center,
        modifier = modifier
            .size(size)
            .clip(CircleShape)
            .background(avatarColor)
            .clearAndSetSemantics { } // Dekorativ – Name steht daneben
    ) {
        Text(
            text = initials,
            color = Color.White,
            fontSize = (size.value * 0.38f).sp,
            fontWeight = FontWeight.Bold
        )
    }
}
